### PR TITLE
[#22] [API] As a user, I can see the details page for a specific keyword

### LINKED ISSUE #22: [API] As a user, I can see the details page for a specific keyword
## Why

Users can view the details of a specific keyword after finishing the process.

## Acceptance Criteria

Create new route and endpoint

### PR DESCRIPTION
**Pull request by vnntsu**
close #22 

## What happened 👀

Create keyword detail API

## Insight 📝

- New route `{GET /api/v1/keyword/{id}}`
- Add unit test

## Proof Of Work 📹

Screenshot 2023-02-07 at 10 20 14
Screenshot 2023-02-07 at 10 37 25

### CODE REVIEW

File: lib/crawler/keyword/keywords.ex [hunk 1]
```diff
@@ -24,6 +24,9 @@ defmodule Crawler.Keyword.Keywords do
   def get_keyword_by_user_id_and_id!(user_id, id),
     do: Repo.get_by!(Keyword, id: id, user_id: user_id)
 
+  def get_keyword_by_user_id_and_id(user_id, id),
+    do: Repo.get_by(Keyword, id: id, user_id: user_id)
```
Comment (andyduong1920) [inline lib/crawler/keyword/keywords.ex:28]: Just FYI, we can use the `get_by!` and let it raise the exception like the Web version, and we could handle the `not_found` template for JSON format. That requires some additional setup in the view, it's out of the IC's scope ==> So I'm ok with the current implementation 👌

File: test/crawler/keyword/keywords_test.exs [hunk 2]
```diff
@@ -106,6 +106,27 @@ defmodule Crawler.Keyword.KeywordsTest do
     end
   end
 
+  describe "get_keyword_by_user_id_and_id/2" do
+    test "given a valid keyword id, returns stored keyword" do
```
Comment (andyduong1920) [inline test/crawler/keyword/keywords_test.exs]:
```suggestion
    test "given a valid keyword id, returns the keyword" do
```

File: test/crawler/keyword/keywords_test.exs [hunk 3]
```diff
@@ -106,6 +106,27 @@ defmodule Crawler.Keyword.KeywordsTest do
...
+      %{id: user_id} = insert(:user)
+      keyword = insert(:keyword, user_id: user_id, name: "keyword")
+
+      assert stored_keyword = Keywords.get_keyword_by_user_id_and_id(user_id, keyword.id)
+      assert stored_keyword.id == keyword.id
+      assert stored_keyword.name == "keyword"
+      assert stored_keyword.user_id == user_id
+    end
+
+    test "given a keyword belongs to another user, returns nil" do
```
Comment (andyduong1920) [inline test/crawler/keyword/keywords_test.exs:167]: let's add the test for `given a non-existing keyword id, returns nil`

Comment (vnntsu) [reply test/crawler/keyword/keywords_test.exs:167]: Added in [3dfd2cd](https://github.com/vnntsu/elixir-ic/pull/47/commits/3dfd2cd51f959f48690b21d0820b5b2345d23323)

**Review by andyduong1920 — APPROVED** [review]

**Review by Nihisil — APPROVED** [review]
